src/lib/priceEncoder.ts

```diff
@@ -0,0 +1,28 @@
+import { BigNumber } from "ethers";
+import bn from "bignumber.js";
+
+bn.config({ EXPONENTIAL_AT: 999999, DECIMAL_PLACES: 18 });
+
+function bigNumberToBig(val: BigNumber, decimals: number): bn {
+  return new bn(val.toString()).div(new bn(10).pow(decimals));
+}
+export function encodePriceSqrt(price: BigNumber): BigNumber {
```

Comment:
price will have to be an integer here, Ethers.BigNumber [does not support decimals](https://github.com/indutny/bn.js#usage).

Reply:
I'm not sure this is a problem, since the number we pass to `BigNumber.from()` is so huge as a result of the rest of manipulations, though I have no idea why we manipulate prices like that, @WarTech9 can you explain a bit please

Reply:
What we're doing is converting from a human readable number to a square root of the prices that uniswap understands. The reference for this is here: https://docs.uniswap.org/sdk/guides/fetching-prices#understanding-sqrtprice
I'll add this to the comments.